### PR TITLE
Refactor away from using Cluster object in nodeup

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

**Comment by johngmyers** [comment]
@justinsb I can't quite figure out the gce-cilium e2e failure. dns-controller can't connect to Gossip. It's possibly related to #14877.

**Comment by johngmyers** [comment]
Ah, my change broke Gossip because nodeup was checking the Name field of the Cluster object.

File: nodeup/pkg/model/context.go [hunk 1]
```diff
@@ -60,14 +61,17 @@ type NodeupModelContext struct {
...
 
 	// IsMaster is true if the InstanceGroup has a role of master (populated by Init)
 	IsMaster bool
 
 	// HasAPIServer is true if the InstanceGroup has a role of master or apiserver (pupulated by Init)
 	HasAPIServer bool
 
+	// IsGossip is true if the cluster runs Gossip DNS.
+	IsGossip bool
+
```
[the remark this answers is not in the document]
Comment (justinsb) [reply nodeup/pkg/model/context.go:74]: I'd also lean towards a function if we can determine it from other values, but ... I don't think it's a blocker as it's only internal implementation (I believe)

**Comment by justinsb** [comment]
Generally lgtm - and I really like the direction of the refactor.  A few quibbles over when we should use a function, but we can maybe reach some guidelines in office hours (if we're having them today?  If not I vote to merge and iterate, I might propose a few helper functions for the hostnames in particular).

I really like the direction here though, I would love to have nodeup only work in terms of Tasks that come from kops-controller (or from metadata in the case of control plane).  That would require us to treat (nodeup) Tasks as versioned, but I think it would be great to simplify nodeup and enable things like "hot" upgrades.  But that's more directional IMO!

**Comment by johngmyers** [comment]
Is there anything else for this PR?

**Review by hakman — APPROVED** [review]

**Pull request merged**